### PR TITLE
[#947] Authorship view: update sort controls

### PR DESCRIPTION
**Pull request by Tejas2805**
Fixes part of #947 

```
The sort controls in the Authorship view are a value updated by
watched property. This is not as efficient as a computed property. 

Let's make the sort controls in Authorship view a computed 
property as it is more efficient than its current implementation. 

This will also help in making the code across different views 
consistent.
```

### CODE REVIEW

File: frontend/src/static/js/v_authorship.js [hunk 1]
```diff
@@ -299,11 +285,18 @@ window.vAuthorship = {
   },
 
   computed: {
+
```
Comment (0blivious) [inline frontend/src/static/js/v_authorship.js]: extra blank line

Comment (Tejas2805) [reply frontend/src/static/js/v_authorship.js]: Seemed to have missed it. Updated.

File: frontend/src/static/js/v_authorship.js [hunk 2]
```diff
@@ -299,11 +285,17 @@ window.vAuthorship = {
   },
 
   computed: {
+    sortFiles() {
+      return (a, b) => (this.toReverseSortFiles ? -1 : 1)
+        * window.comparator(filesSortDict[this.filesSortType])(a, b);
+    },
+
```
Comment (0blivious) [inline frontend/src/static/js/v_authorship.js]: Should standardise the name also. This one should be named `sortingFunction`.

**Comment by Tejas2805** [comment]
> LGTM, but the reason in your commit message is actually not the major one. We use computed value is mainly because it is more efficient.

Have updated the commit message accordingly.

**Comment by jamessspanggg** [comment]
Hmm. Not sure if it's related to this PR but when files are sorted by `fileType`, the following occurs where `java` test files are not in order...

Screen Shot 2020-02-15 at 2 37 37 PM

[Link to reproduce](https://deploy-preview-1074--reposense.netlify.com/#search=&sort=groupTitle&sortWithin=title&since=2017-10-09&timeframe=commit&mergegroup=false&groupSelect=groupByRepos&breakdown=false&tabOpen=true&tabType=authorship&tabAuthor=AdityaA1998&tabRepo=reposense%2FRepoSense%5Bmaster%5D)

**Comment by Tejas2805** [comment]
Don't think it is related to this PR. Maybe we are taking the wrong thing into consideration when considering `Filetype`. The rest work all fine.

**Comment by Tejas2805** [comment]
> Don't think it is related to this PR. Maybe we are taking the wrong thing into consideration when considering `Filetype`. The rest work all fine.

Opening a new issue for this.

File: frontend/src/static/js/v_authorship.js (hunk 1 above)
Comment (0blivious) [reply frontend/src/static/js/v_authorship.js]: The extra blank line appear again.

**Comment by Tejas2805** [comment]
@fzdy1914 any update required?

**Pull request merged**